docs/src/2.5.AddingInteractivity.mdx

```diff
@@ -0,0 +1,117 @@
+
+```js
+const [clickedObjectIds, setClickedObjectIds] = useState(new Set());
+
+const obstacleMarkers = Array.from(clickedObjectIds).map((clickedObjectId, index) => {
+  const pointIdx = clickedObjectId - marker.id;
+  const position = marker.points[pointIdx];
+  return {
+    //keep the id unique by adding sphere's id range as offset: marker.id + steps
+    id: marker.id + steps + index,
+    // remember the original clickedObjectId so when the obstacle is clicked, we can
+    // remove the obstacle quickly by update clickedObjectIds
+    clickedObjectId,
+    pose: {
+      orientation: { x: 0, y: 0, z: 0, w: 1 },
+      position: position,
+    },
+    color: { r: 1, g: 0, b: 0, a: 1 }, // red
+    scale: { x: 4, y: 4, z: 4 }, // scale up a little so it's bigger than the spheres
+  };
+});
+```
+
+Now, let's use the clicked Sphere id to update the `clickedObjectIds` state, which will be consumed by the Cube obstacles. Clicking each obstacle should remove it from the scene.
+
+```js
+ <Spheres
+    onClick={(ev, { objectId, object }) => {
+        // Click once to add obstacle. Even though both the small Sphere and the big Obstacle cube have
+        // hitmap enabled, clicking again will only trigger `onClick` event on the Cube because it's a lot bigger
+        clickedObjectIds.add(objectId);
+        setClickedObjectIds(new Set(clickedObjectIds));
+    }}>
+    {[marker]}
+    </Spheres>
+    <Cubes
+        onClick={(ev, { objectId, object }) => {
+            // the `object` contains information of the clicked Cube, which has `clickedObjectId`
+            clickedObjectIds.delete(object.clickedObjectId);
+            setClickedObjectIds(new Set(clickedObjectIds));
+        }}>
+        {obstacleMarkers}
+    </Cubes>
+```
+
+Lastly, setting an `isDuckStopped` variable will allow us to tell the duck when to stop and when to move forward. 
+
+```js
+// keep track whethere we need to stop the duck from moving
+let isDuckStopped = false;
+useRequestAnimationFrame(() => {
+  if (!isDuckStopped) {
+    // update count before each browser repaint
+    const newCount = count >= 100000 ? 0 : count + 1;
+    setCount(newCount);
+  }
+});
+
+obstacleMarkers.forEach((marker) => {
+  const position = marker.pose.position;
+  // make the duck stop when the duck's current position has an obstacle
+  if (position.x === duckPosition.x && position.y === duckPosition.y && position.z === duckPosition.z) {
+    isDuckStopped = true;
+  }
+});
+s;
+```
+
+Put it all together, and we now have a moving duck that gets stuck on every Cube obstacle in its path. Clicking a blocking Cube will remove it, allowing the duck to happily continue forward.
+
+<CodeSandboxEmbed src="https://codesandbox.io/s/j7n5r2mkr3" />
+
+## Next Steps
+
+Congratulations on finishing this tutorial, and building your first  `regl-worldview` app! To learn more:
```

Comment:
Oops – this extra space might have been me.

Reply:
not sure there is any extra space...